text/0017-message/0017-message.md

```diff
@@ -0,0 +1,181 @@
+    In order to parse it we look at its bits: 
+    `1000 1111 1000 0000 0000 0000`.
+    First we strip the msbs:
+    `000 1111 000 0000 000 0000`
+    Then we reverse the groups of 7 bits (Little Endian):
+    `000 0000 000 0000 000 1111`
+    Then we concatenate all the bits to finally get:
+    `1111`
+    So the varint `0x0F` equals to `0x8F8000`, but is encoded with different bytes.
+    * *Size Protection*: The varint must not be more than 10 bytes long. A 10 byte varint can encompass all the range of `uint64`. In case of an overflow above the `uint64` range the message is invalid.
+3. When we are done parsing the message there shouldn't be any trailing bytes left that were not parsed.
+4. If the `payload type` is in the core payload range (0-127) and the node is familiar with it, or if it is in the range of 128-16383. The upper limit of 16383 is the maximal value of a 2 byte varint, and it should suffice for all the optional payload types we need.
+5. If the Message PoW Hash will contain at least the number of trailing 0 trits the node defines as required.
+
+### Payloads
+
+Each message will contain a payload. The specification of the payloads is out of scope of this RFC. Below is a table of the currently specified core payloads with a link to their specifications. The `Unsigned Payload` will be specified here as an example.
+
+| Payload Name                              |   Type Value |
+| ---------------------------------------   | -----------  | 
+|  Signed Transaction                       |     0        |
+|  Milestone                                |     1        |
+|  [Unsigned Data](#Unsigned-Data-Payload)  |     2        |
+| Signed Data                               |     3        |
+| Indexation                                |     4        |
+
+### Unsigned Data Payload
+
+To make the Payload concept clear we will define the `unsigned data payload`. As the name suggests it simply allows to add arbitrary data to a message. A message that has been attached to the tangle and approved by a milestone has useful properties: You can verify that the content of the data did not change, and you can ascertain the approximate time it was published by checking the approving milestone. For example, one can record a hash of a text document on the tangle with such a payload. When a third party requests a proof that the document was published at a certain time and was not changed, one can point them to them to the message containing the `unsigned data payload` with the hash.
+
+The structure of the payload is simple:
+
+
+
+| Name             | Type     | Description               |
+| --------         | -------- | -----------               |
+| Payload Type     | varint   | Must be set to **2**      |
+| Data             | ByteArray| The data we are attaching |
+
+There are no validation rules for the payload. Message validation rules suffice here.
```

Comment:
Does this mean that an `Unsigned Data Payload` with a byte array of length zero is valid?

Reply:
If the `payload length` field in the message accommodates only for the type then in my opinion it should be valid.

Obviously there is no point in creating such a payload, but I don't see an issue here